project/MiMa.scala

```diff
@@ -87,6 +87,19 @@ object MiMa extends AutoPlugin {
       ProblemFilters.exclude[DirectMissingMethodProblem]("akka.cluster.ClusterCoreDaemon.removed"),
       ProblemFilters.exclude[DirectMissingMethodProblem]("akka.cluster.Gossip.convergence"),  
         
+      //#21717 Improvements to AbstractActor API
+      FilterAnyProblemStartingWith("akka.japi.pf.ReceiveBuilder"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.AbstractActor.receive"),
+      ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.actor.AbstractActor.initialReceive"),
+      ProblemFilters.exclude[MissingClassProblem]("akka.actor.AbstractActorContext"),
+      ProblemFilters.exclude[IncompatibleResultTypeProblem]("akka.actor.AbstractActor.getContext"),
+      ProblemFilters.exclude[IncompatibleResultTypeProblem]("akka.actor.AbstractActor.emptyBehavior"),
+      ProblemFilters.exclude[MissingTypesProblem]("akka.actor.ActorCell"),
```

Comment:
I'm somewhat scared of this change, that is due to the move of AbstractActorContext. Could it break bin compat for scala actors. I think I'll have to do a real test with an example application.